src/langchain_google_spanner/vector_store.py

```diff
@@ -832,15 +967,111 @@ def similarity_search_with_score_by_vector(
             List[Document]: List of documents most similar to the query.
         """
 
-        results, column_order_map = self._get_rows_by_similarity_search(
+        results, column_order_map = self._get_rows_by_similarity_search_knn(
             embedding, k, pre_filter
         )
         documents = self._get_documents_from_query_results(
             list(results), column_order_map
         )
         return documents
 
-    def _get_rows_by_similarity_search(
+    def search_by_ANN(
```

Comment:
You need to align your functions with [VectorStore](https://api.python.langchain.com/en/latest/vectorstores/langchain_core.vectorstores.VectorStore.html) interface defined in langchain. 

Based on the `QueryParameters.NearestNeighborsAlgorithm` which user is passing at the time of initialization of this class `not init_vector_store_table`, we should decide whether we want to use the KNN/ANN.

Reply:
Indeed. Done.